components/atoms/RemoveItem.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react'
+import { Component }  from 'react'
+import styled from 'styled-components'
+import Link from 'next/link'
+import Img from 'react-image'
+import List from './List'
+
+
+const ImgWrapper = styled.span`
+   /* visibility: hidden; */
+
+/* &:hover {
+    visibility: visible;
+    display:inline-block;
+} */
+
+`;
+
+
+export default class RemoveItem extends Component {
+    constructor (props){
+        super(props);
+        this.removeIndex = null
+    }
+    render() {
+       return(
+        
+            <ImgWrapper className={this.props.className}>
+                <img src="../../static/images/trash-icon.png" alt="Remove" height="25" style={{paddingLeft:'20px'}}
+                    ref={img => this.removeIndex = img}
+                    onClick={e => {this.props.onRemove(this.removeIndex)}}
```

Comment:
This `ref` thing seems a bit off.  A `ref` is a reference to the underlying HTML element, which you are then passing along to the `removeIndex` function.  I assume you're expecting this to be a number that corresponds to the items index in the array, but if you console.log this out it won't be a number.  

Perhaps you can pass the index as a `prop` to this component from the parent?